gcpkms/keysource.go

```diff
@@ -131,12 +131,15 @@ func (key MasterKey) createCloudKMSService() (*cloudkms.Service, error) {
 	}
 
 	ctx := context.Background()
-	client, err := google.DefaultClient(ctx, cloudkms.CloudPlatformScope)
-	if err != nil {
+	var options []option.ClientOption
+
+	if credentials, err := getGoogleCredentials(); err != nil {
 		return nil, err
+	} else if len(credentials) > 0 {
+		options = append(options, option.WithCredentialsJSON(credentials))
```

Comment:
Setting client option only if a credential is returned from `GOOGLE_CREDENTIALS`. Otherwise, default to using default application credentials as defined in the [`cloudkms.NewService()`](https://github.com/googleapis/google-api-go-client/blob/5ba22f18bcf26dbbf4cd1432504269141274170f/cloudkms/v1/cloudkms-gen.go#L17-L22)